inginious/frontend/plugins/user_management/pages/api/user_data.py

```
@@ -0,0 +1,139 @@
+            flag = True
+            collection_name_list = json.loads(get_mandatory_parameter(user_data, "collection_list"))
+            username_count = change_username(username, user_data["new_username"], collections_manager,
+                                             collection_name_list)
+        if not flag:
+            return 400, {"error": _("no data to change")}
+
+        new_username = user_data["new_username"] if username_count > 0 else username
+        user_final_info = self.get_user_data(new_username)
+        make_user_changes_register(user_original_info, user_final_info, collections_manager)
+
+        remove_block_user(new_username, collections_manager)
+
+        try:
+            inform_user_changes(user_original_info, user_final_info, collections_manager)
+        except api.APIError as error:
+            return error.status_code, {"username": username_count, "email": email_count, "name": name_count,
+                                       "error": error.return_value}
+
+        return 200, {"username": username_count, "email": email_count, "name": name_count}
+
+    def get_user_data(self, username):
+        """ Returns data of a user """
+        collections_manager = CollectionsManagerSingleton.get_instance()
+        user_basic_data = self.database.users.find_one({'username': username})
+        if user_basic_data:
+            collection_data, unknown_collections = get_count_username_occurrences(user_basic_data["username"],
+                                                                                  collections_manager)
+            data = {"username": user_basic_data["username"],
+                    "name": user_basic_data["realname"],
+                    "email": user_basic_data["email"],
+                    "count": collection_data,
+                    "unknown_collections": unknown_collections}
+            return data
+        else:
+            return {"user": _("User no found")}
```

Probably raise an exception rather than returning this object. Something like `raise api.APIError(error_code, _("User no found"))` and then you can catch the exception when this method is called.